lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2752,6 +2793,21 @@ private static function get_block_nodes( $theme_json, $selectors = array(), $opt
 				$nodes[] = array(
 					'path' => $node_path,
 				);
+
+				if ( $include_variations && isset( $node['variations'] ) ) {
+					foreach ( $node['variations'] as $variation => $node ) {
+						$nodes[] = array(
+							'path' => array( 'styles', 'blocks', $name, 'variations', $variation ),
+						);
+						if ( isset( $node['blocks'] ) ) {
```

Comment:
Check with @aaronrobertshaw about whether this is all correct 😄

Reply:
It appears to match the original code in the else statement. I'm missing context here though around the inclusion of `include_node_paths_only`. It looks like the variations part of the equation was mistakenly overlooked during that enhancement.

One thing I haven't grokked yet is where the paths get handled for element styles within block style variations. Were they covered?

Reply:
> One thing I haven't grokked yet is where the paths get handled for element styles within block style variations. Were they covered?

No not covered. They'll need to be added. I deliberately skipped over them, because I wasn't sure whether it was appropriate to overload `get_block_nodes()` in that way when they didn't previously deal with nested block styles or elements in block style variations.

TBD I guess.


> It looks like the variations part of the equation was mistakenly overlooked during that enhancement.

That's what I thought - I moved it up to make this PR work, but I suppose it should be a separate PR.

Fortunately it's only used in one place for now `WP_Theme_JSON:merge` and only in the context of merging background images. So I don't think there's any urgency to it.